core/types/receipt.go

```diff
@@ -28,10 +28,16 @@ import (
 
 //go:generate gencodec -type Receipt -field-override receiptMarshaling -out gen_receipt_json.go
 
+const (
+	ReceiptStatusSuccess = byte(0x01)
+	ReceiptStatusFailure = byte(0x00)
+)
+
 // Receipt represents the results of a transaction.
 type Receipt struct {
 	// Consensus fields
 	PostState         []byte   `json:"root"`
+	Status            byte     `json:"status"`
```

Comment:
Since the status can only ever be "failed" or "succeeded", it would make more sense to me to keep this field as a `bool` called `Failed`. The actual serialization will of course remain the `0x00` or `0x01` byte, but it seems cleaner to work with a boolean in Go code if nothing else is possible (or will be in the planned future).